sdk/core/azure-core/inc/azure/core/io/body_stream.hpp

```diff
@@ -282,4 +283,36 @@ namespace Azure { namespace Core { namespace IO {
     int64_t Length() const override;
   };
 
+  /**
+   * @brief A concrete implementation of #Azure::Core::IO::BodyStream that wraps another stream and
+   * reports progress
+   */
+  class ProgressBodyStream : public BodyStream {
+  private:
+    BodyStream& m_bodyStream;
+    int64_t m_bytesTransferred;
+    std::function<void(int64_t bytesTransferred)> m_callback;
+
+  private:
+    size_t OnRead(uint8_t* buffer, size_t count, Azure::Core::Context const& context) override;
+
+  public:
+    /**
+     * @brief Constructs `%ProgressBodyStream` from a %BodyStream.
+     *
+     * @param bodyStream the body stream to wrap
```

Comment:
nit: Use complete sentences for doc strings (these are used to generate API reference docs using doxygen)

```suggestion
     * @param bodyStream The body stream to wrap.
```

Reply:
done